### PR TITLE
chore: support python version `3.10`

### LINKED ISSUE #795: Support Python 3.10
**Describe the bug/feature**
Support Python 3.10

There might be issues with reclass cc @simu 

to be updated..

### PR DESCRIPTION
**Pull request by MatteoVoges**
Fixes #795 

## Proposed Changes
- add python3.10 to tests
- cleanup artefacts from python3.7

--- 
### Contributed by  [](https://www.nexenio.com)

### CODE REVIEW

File: .github/workflows/pex-build-upload.yml [hunk 1]
```diff
@@ -33,7 +33,7 @@ jobs:
       - name: Build Pex
         run: |-
           mkdir -p dist
-          pex .[gojsonnet] -r requirements.txt --python-shebang='#!/usr/bin/env python3' --python=python3.7 --python=python3.8 --python=python3.9 -m kapitan -o dist/kapitan.linux-x86_64.pex
+          pex .[gojsonnet] -r requirements.txt --python-shebang='#!/usr/bin/env python3' --python=python3.8 --python=python3.9 --python=python3.10 -m kapitan -o dist/kapitan.linux-x86_64.pex
```
Comment (jkrzemin) [inline .github/workflows/pex-build-upload.yml:36]: Are we dropping testing for python 3.7 then?

Comment (MatteoVoges) [reply .github/workflows/pex-build-upload.yml:36]: It has been already dropped, but we forgot to change it here. I think the action isn't even used anymore, but just I wanted to keep it up to date for potential future reuse.

File: README.md [hunk 2]
```diff
@@ -32,16 +32,25 @@ On Linux you can add `-u $(id -u)` to `docker run` to preserve file permissions.
 
 ### Pip
 
-Kapitan needs Python 3.7.
+Kapitan needs Python 3.8.
```
Comment (antonincms) [inline README.md:35]: Could this be reformulated to show kapitan support Python versions 3.8 up to 3.10 ?

Comment (MatteoVoges) [reply README.md:35]: Good idea

Comment (antonincms) [reply README.md:35]: In which case here is a suggestion : 
```suggestion
Kapitan supports Python 3.8, 3.9 and 3.10.
```

**Pull request closed without merging**